sound/soc/sof/intel/hda-dai.c

```diff
@@ -451,5 +463,33 @@ struct snd_soc_dai_driver skl_dai[] = {
 	.name = "Alt Analog CPU DAI",
 	.ops = &hda_link_dai_ops,
 },
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_DEBUG_PROBES)
+{
+	.name = "Probe Injection0 CPU DAI",
+	.compress_new = snd_soc_new_compress,
+	.cops = &sof_probe_compr_ops,
+	.playback = {
+		.stream_name = "Probe Injection0",
+		.channels_min = 1,
+		.channels_max = 8,
+		.rates = SNDRV_PCM_RATE_48000,
+		.rate_min = 48000,
+		.rate_max = 48000,
+	},
+},
+{
+	.name = "Probe Extraction CPU DAI",
+	.compress_new = snd_soc_new_compress,
+	.cops = &sof_probe_compr_ops,
+	.capture = {
+		.stream_name = "Probe Extraction",
+		.channels_min = 1,
+		.channels_max = 8,
+		.rates = SNDRV_PCM_RATE_48000,
+		.rate_min = 48000,
+		.rate_max = 48000,
+	},
+},
+#endif
```

Comment:
should these DAIs really be added with hardware related ones?

This feels like they could be added with topology, not sure why they need to be hard-coded.

Reply:
Also the comments in the commit message are quite misleading.
The probes are intended to be host-facing interfaces, so the moral equivalent of a FE. You will not need to have a FE tied to a compressed DAI, and that might not even make sense from a technical perspective.

Reply:
Same answer as for the similar comment below: https://github.com/thesofproject/linux/pull/1276#discussion_r332972439

Does not mean I'm not up for the discussion. Currently, code expresses my point of view.

Reply:
I think reserving those DAIs for a 'probe client' in an MFD solution would be the right thing to do.